<!doctype html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <title>ParSe v2 - FASTA</title>
    <meta name="description" content="">

    <link href="https://cdn.jsdelivr.net/npm/normalize.css@8.0.1/normalize.css" rel="stylesheet">
    <link href="https://unpkg.com/tabulator-tables@5.4.3/dist/css/tabulator.min.css" rel="stylesheet">
    <link rel="stylesheet" href="./css/app.css">
</head>

<body>
    <div id="loading" class="loading-overlay">
        <div id="loading-text" class="loading-text">Loading</div>
    </div>
    <main>
        <h1>ParSe v2: predict phase-separating protein regions from the primary sequence</h1>
        <p>This is a form that allows you to enter a set of sequences and quickly find those predicted by the algorithm to drive protein-mediated physiological phase separation. The algorithm also determines if the set as a whole is enriched or depleted in phase separation potential compared to the reference human proteome, obtained from <a href="https://www.uniprot.org/">UniProt</a> during summer 2022.</p>
        <p>To analyze a single protein sequence using ParSe v2, please visit <a href="https://stevewhitten.github.io/Parse_v2_web/">our other site</a>.</p>
        <p>The algorithm is described in Ibrahim et al, <a href="https://doi.org/10.1016/j.jbc.2022.102801"><i>J. Biol. Chem.</i> <b>299</b>, 102801 (2023)</a>.</p>
        <h3>Brief Description</h3>
        <p>ParSe v2 explores the possibility that the potential to undergo liquid-liquid phase separation (LLPS) can be predicted from sequence-based calculations of
            hydrophobicity, α-helix propensity, and a model of the polymer scaling exponent (ν<sub><i>model</i></sub>).
            Using these three factors, any protein sequence can be parsed into one
            of three categories:</p>
        <ul>
            <li><b>P Regions (P)</b> are intrinsically disordered and prone
                to undergo LLPS.</li>
            <li><b>D Regions (D)</b> are intrinsically disordered but do not
                undergo LLPS.</li>
            <li><b>F Regions (F)</b> may or may not be intrinsically disordered,
                but can fold to a stable conformation.</li>
        </ul>
        <p>The computed “PS potential” of a sequence is the sum of a parameter called the “classifier distance” over all positions labeled P.</p>
        <p>We have found that homotypic phase-separating (PS) proteins generally have computed PS potentials greater than 100, and predicted PS IDRs 50 residues in length or longer.</p>
        <div class="mb-10">
            <h3>Sequence File</h3>
            <p>To use this site, input a FASTA file containing protein sequences. Files with tens of thousands of sequences may take a minute or longer to analyze. The algorithm ignores sequences longer than 10,000 residues or shorter than 25 residues. Amino acids are restricted to the 20 common types.
            </p>
            <p>After the analysis has finished, sort the table of sequence-calculated properties by the third column to rank sequences by computed PS potential, or by the fourth column to rank by length of the longest predicted PS IDR. The last three columns give the UniProt accession number, gene and protein names if provided in the input FASTA file, i.e. >sp|UniProt|gene protein.</p>
            <div class="mb-10">
                <label for="input-file">File: </label>
                <input id="input-file" type="file" accept=".txt, .fasta, .fasta.gz" />
            </div>
            <div class="mb-10">
                <label for="proteome-name">Name for the file (optional): </label>
                <br/>
                <input id="proteome-name" />
            </div>
            <div class="mb-10">
                <button id="submit">Submit</button>
            </div>
        </div>
        <h3>A</h3>
        <p class="mb-10" id="auc-a"></p>
        <div class="mb-10">
            <div class="chart-container">
                <div class="mb-10 chart" id="chart-a"></div>
                <div class="mb-10 chart" id="chart-auc-a"></div>
            </div>
            <p><b>A. Distribution of predicted PS IDR lengths.</b> ParSe v2 finds regions that were ≥90% labeled P, which are referred to as phase-separating, PS, regions. The y-axis is the percent of proteins in a set with PS regions at least as long as the length indicated by the x-axis. The recall plot directly compares the percent of set for the human proteome against itself and against the input file.</p>
            <h3>B</h3>
            <p class="mb-10" id="auc-b"></p>
            <div class="chart-container">
                <div class="mb-10 chart" id="chart-b"></div>
                <div class="mb-10 chart" id="chart-auc-b"></div>
            </div>
            <p><b>B. Distribution of PS potentials.</b> ParSe v2 calculates the classifier distance sum of P-labeled positions, which we use as a numerical score to estimate the PS potential. The y-axis is the percent of proteins in a set with a classifier distance sum of P-labeled positions at least as much as the value indicated by the x-axis. The recall plot directly compares the percent of set for the human proteome against itself and against the input file.</p>
            <div class="mb-10">
            <label for="download-summary-csv-b">Sequence-calculated properties: </label>
                <button id="download-summary-csv-b">Download CSV</button>
            </div>
            <div class="mb-10" id="summary-b"></div>
            <div class="mb-10">
            <label for="download-data-csv-a">A. Percent of set and recall plot: </label>
                <button id="download-data-csv-a">Download CSV</button>
            </div>
            <div class="mb-10" id="data-a"></div>
        </div>
        <div class="mb-10">
            <div class="mb-10">
            <label for="download-data-csv-b">B. Percent of set and recall plot: </label>
                <button id="download-data-csv-b">Download CSV</button>
            </div>
            <div class="mb-10" id="data-b"></div>
        </div>
    </main>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/pako/2.0.4/pako.min.js"
        integrity="sha512-EAKzSKex+PXC0U9OG13r1059ysjrjkJEpZoONCnZa0mBROY28iBOOxZSErUVw1LzLr2+U5PhR7zPCPKidUVJqg=="
        crossorigin="anonymous" referrerpolicy="no-referrer"></script>
    <script src="https://cdn.plot.ly/plotly-2.4.2.min.js"></script>
    <script type="text/javascript" src="https://unpkg.com/tabulator-tables@5.4.3/dist/js/tabulator.min.js"></script>
    <script src="./js/part-a.js"></script>
    <script src="./js/part-b.js"></script>
    <script src="./js/constants.js"></script>
    <script src="./js/parse.js"></script>
    <script src="./js/app.js"></script>
</body>

</html>
